sympy/physics/continuum_mechanics/cable.py

```diff
@@ -0,0 +1,334 @@
+
+    Cables are widely used in suspension bridges, tension
+    leg offshore platforms, transmission lines, and find
+    use in several other engineering applications.
+    """
+    def __init__(self):
+        """
+        Initializes the class.
+        """
+        self._left_support = []
+        self._right_support = []
+        self._supports = {}
+        self._support_labels = []
+        self._loads = {}
+        self._loads_position = {}
+        self._length = 0
+        self._reaction_loads = {}
+
+    @property
+    def supports(self):
+        """
+        Returns the supports of the cable along with their
+        positions.
+        """
+        return self._supports
+
+    @property
+    def left_support(self):
+        """
+        Returns the position of the left support.
+        """
+        return self._left_support
+
+    @property
+    def right_support(self):
+        """
+        Returns the position of the right support.
+        """
+        return self._right_support
+
+    @property
+    def loads(self):
+        """
+        Returns the magnitude and direction of the loads
+        acting on the cable.
+        """
+        return self._loads
+
+    @property
+    def loads_position(self):
+        """
+        Returns the position of the loads acting on the
+        cable.
+        """
+        return self._loads_position
+
+    @property
+    def length(self):
+        """
+        Returns the length of the cable.
+        """
+        return self._length
+
+    @property
+    def reaction_loads(self):
+        """
+        Returns the reaction forces at the supports.
+        """
+        return self._reaction_loads
+
+    def add_supports(self, *args):
+        """
+        This method adds the right and left supports to the Cable object.
+
+        Parameters
+        ==========
+        Pass two tuples of the form (label, x, y) as inputs, where the first
+        tuple represents the left support, while the later represents the
+        right support. Each element in these tuples are:
+
+        label: String or a symbol
+            The label for the support.
+
+        x: Sympifyable
+            The x-coordinate of the position of the support.
+
+        y: Sympifyable
+            The y-coordinate of the position of the support.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.continuum_mechanics.cable import Cable
+        >>> c = Cable()
+        >>> c.add_supports(('A', 3, 4), ('B', 4, 5))
+        >>> c.supports
+        {'A': [3, 4], 'B': [4, 5]}
+        >>> c.left_support
+        [3, 4]
+        >>> c.right_support
+        [4, 5]
+        """
+        if len(self._supports) != 0:
+            raise ValueError("Supports already exist")
+
+        elif len(args) != 2:
+            raise ValueError("Pass only two arguments: left support and right support")
+
+        elif args[0][0] == args[1][0]:
+            raise ValueError("Supports can not have the same label")
```

Comment:
While applying a support you should also check if a support of same label exists elsewhere